### ISSUE TITLE
First run `Docker-compose up` fails with macOS version of Docker

### ISSUE THREAD

**Issue by dlindberg** [nystudio107/craft#33]
### Work around / possible solution

Before running `docker-compose up` the first time, create the following directories:
```
mkdir docker-config/webpack-dev-craft/node_modules
mkdir cms/vendor
mkdir cms/web/dist
mkdir cms/storage
mkdir cms/storage/logs
```

### Describe the bug
When the `delegated` flag is included on a bind mount docker fails to create a directory in the host file system and throws errors for building the `webpack` and `php` containers. This issue seems like it may be an issue with Docker and not the project itself (see additional troubleshooting below).

### To reproduce
Steps to reproduce the behaviour:
1. `composer create-project nystudio107/craft docker-example`
2. `cd docker-example`
3. `docker-compose up`
4. error, see screen shot.

### Expected behavior
Get to the first run issue described in the readme. Creating the above directories allows the containers to successfully build to the documented point where `docker-example_php_1 exited with code 1` and then continue the first run process successfully as described.

### Screenshots
![image](https://user-images.githubusercontent.com/684461/86371463-e4c7ed00-bc4e-11ea-847e-01aa146f8d8e.png)

### Additional troubleshooting / Investigation
Running: `docker run -d -it --name bad-mount-delegated -v "$(pwd)"/nodir:/app:delegated nginx:latest` results in the error: `docker: Error response from daemon: unable to enable two-way sync on [path]/docker-example/nodir: file does not exist. See 'docker run --help'.`

However, simply running `docker run -d -it --name bad-mount -v "$(pwd)"/notdir:/app nginx:latest` successfully spins up the container and creates the needed directory.

Issue #2781 on the docker compose repo says the short syntax is supposed to use `-v` to bind volumes whereas the long syntax is supposed to use `--mount` [which differ on the response when there is no directory present](https://github.com/docker/compose/issues/2781#issuecomment-501654001). It seems like using the `:delegated` flag is the source of the problem. I couldn't find anything that suggested using performance flags overrides the `-v` bind style in the documentation. Given the specific error cites a sync problem that seems time out I looked into a possible [permissions issue with macOS TCC](https://objekt.click/2019/11/docker-the-problem-with-macos-catalina/). However, I was unable to resolve the issue by adjusting permissions. Ultimately, I wasn't able to determine if what is happening with docker and the OS is `behaves as expected` or not. However, there was a significant amount of discussion on the issue thread that allowing the -v option to create a directory where none exists can result in unexpected behaviors and unexpected permissions on the created directories. So it may be preferable to have the project handle the directory creation explicitly instead of allowing Docker to handle it implicitly even if it is supposed to be doing so.

### Versions
- Docker desktop version: 2.3.2.0 (46268)
- Docker Engine version: 19.03.12
- macOS version: 10.15.5 (Catalina)

**Comment by khalwat** [comment]
related: https://github.com/nystudio107/craft/pull/28

related: https://github.com/craftcms/cms/issues/6729

**Comment by khalwat** [comment]
This is fixed now in the 2.4.x release, with a seed db and other checks
